### ISSUE TITLE
`cosign copy` ignores `--attachment-tag-prefix`

### ISSUE THREAD

**Issue by znewman01** [sigstore/cosign#2962]
```shell
$ SRC_IMAGE=alpine
$ SRC_DIGEST=$(crane digest $SRC_IMAGE)
$ IMAGE_URI=ttl.sh/$(uuidgen | head -c 8 | tr 'A-Z' 'a-z')
$ crane copy $SRC_IMAGE@$SRC_DIGEST $IMAGE_URI:1h
$ IMAGE_URI_DIGEST=$IMAGE_URI@$SRC_DIGEST
$ cosign sign --attachment-tag-prefix-demo-sign- $IMAGE_URI_DIGEST
$ crane ls $IMAGE_URI
1h
demo-sign-sha256-124c7d2707904eea7431fffe91522a01e5a861a624ee31d03372cc1d138a3126.sig
$ IMAGE_URI2=ttl.sh/$(uuidgen | head -c 8 | tr 'A-Z' 'a-z')
$ cosign copy --attachment-tag-prefix demo-sign- $IMAGE_URI_DIGEST $IMAGE_URI2
$ crane ls $IMAGE_URI2
latest
```

**Version**

```
$ cosign version
  ______   ______        _______. __    _______ .__   __.
 /      | /  __  \      /       ||  |  /  _____||  \ |  |
|  ,----'|  |  |  |    |   (----`|  | |  |  __  |   \|  |
|  |     |  |  |  |     \   \    |  | |  | |_ | |  . `  |
|  `----.|  `--'  | .----)   |   |  | |  |__| | |  |\   |
 \______| \______/  |_______/    |__|  \______| |__| \__|
cosign: A tool for Container Signing, Verification and Storage in an OCI registry.

GitVersion:    v2.0.0
GitCommit:     unknown
GitTreeState:  clean
BuildDate:     unknown
GoVersion:     go1.20.2
Compiler:      gc
Platform:      darwin/arm64
```

**Comment by thangarajav** [comment]
Any update on this issue

**Comment by mritunjaysharma394** [comment]
Tried to fix it through PR #3014, I guess it was happening because `regOpts.ClientOpts` wasn't being called here which actually handles the underlying logic of the `attachment-tag-prefix` flag. Looking forward to the feedback!
